src/SFA.DAS.AssessorService.Application.UnitTests/Handlers/Certificates/StartCertificateHandlerTests/When_called_and_no_existing_certificate.cs

```diff
@@ -25,16 +26,17 @@ public class When_called_and_no_existing_certificate
         private StartCertificateHandler _startCertificateHandler;
         private Guid _organisationId;
         private Certificate _returnedCertificate;
-        private Certificate _savedCertificate;
+        private List<Certificate> _savedCertificates;
 
         [SetUp]
         public void Arrange()
         {
             _certificateRepository = new Mock<ICertificateRepository>();
             _certificateRepository.Setup(r => r.GetCertificate(1111111111, 30)).ReturnsAsync(default(Certificate));
 
+            _savedCertificates = new List<Certificate>();
             _certificateRepository.Setup(r => r.New(It.IsAny<Certificate>())).ReturnsAsync(new Certificate() { CertificateReferenceId = 10000 })
-                .Callback<Certificate>(c => _savedCertificate = c);
+                .Callback<Certificate>(c => _savedCertificates.Add(c));
 
```

Comment:
Just wondering what this setup .Callback does?

Reply:
The Callback is invoked before or after the return depending on when it's added to the chain. The ideas is that if you wanted to do additional action in that window you are able to. In this case I wanted to capture the object passed into the method. I find it easier to read to save this object and assert on it, than a long winded verify though i think i've changed this now with the test rework

Reply:
Thanks Chris!